File: docs/guides/go/src/globalAlgoliaUserToken.go
package main

import (
	"fmt"

	"github.com/algolia/algoliasearch-client-go/v4/algolia/search"
)

func globalAlgoliaUserToken() {
	client, err := search.NewClient("YourApplicationID", "YourAdminAPIKey")
	if err != nil {
		fmt.Println(err)
	}

	client.AddDefaultHeader("X-Algolia-UserToken", "test-user-123")
}

File: docs/guides/java/src/test/java/com/algolia/globalAlgoliaUserToken.java
package com.algolia;

import com.algolia.api.SearchClient;
import com.algolia.config.*;
import com.algolia.model.search.*;

public class globalAlgoliaUserToken {

  public static void main(String[] args) throws Exception {
    SearchClient client = new SearchClient(
      "ALGOLIA_APPLICATION_ID",
      "ALGOLIA_API_KEY",
      ClientOptions.builder().addDefaultHeader("X-Algolia-UserToken", "test-user-123").build()
    );
    client.close();
  }
}
